### PR TITLE
[New Rule] Azure Storage Account Key Regenerated

### LINKED ISSUE #176: [New Rule] Azure Storage Account Key Regenerated
## Description
Identifies a rotation to your storage account access keys in Azure. Regenerating access keys can affect any applications or Azure services that are dependent on the storage account key. 

## False Positive Info
It's recommended that you rotate your access keys periodically to help keep your storage account secure. Normal key rotation can be exempted from the rule. An abnormal time frame and/or a key rotation from unfamiliar users, hosts, or locations should be investigated.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`azure`

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal

## Syntax
```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:MICROSOFT.STORAGE/STORAGEACCOUNTS/REGENERATEKEY/ACTION and event.outcome:Success
```

## Example Data

image

### PR DESCRIPTION
**Pull request by bm11100**
## Issue(s)
resolves #176 

## Description

Identifies a rotation to storage account access keys in Azure. Regenerating access keys can affect any applications or Azure services that are dependent on the storage account key.

## Checklists
Use ~~strikethroughs~~ to remove items which are not applicable to this issue.

### For submitter
- [x] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
  -- Query field process.name.text doesn't currently exist in endgame-* eventing data, but does exist for winlogbeats, and is case insensitive.
- [x] Verify all query fields co-exist within a single ECS version and annotated the minimum version
  -- 1.4.0
- [ ] Verified the query detects the intended event(s)
  - detonate
  - search in _discover_
  - trigger as a custom signal
- [x] Create rule using `create-rule`
- [x] Verify and convert to standard linting (`toml-lint -f <rule-file>`)
- [x] Run tests using `pytest -x -v unit_tests` or `make test`
- [ ] Internal search to determine noise and FP

### For reviewers
- [ ] Verify existing rule for activity doesn't exist as a siem or endpoint rule (unless intentionally specified in issue)
- [ ] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
- [ ] Verify all query fields co-exist within a single ECS version and annotated the minimum version
- [ ] Internal search to determine noise and FP
- [ ] Verify metadata accuracy and spelling

### CODE REVIEW

File: rules/azure/credential_access_storage_account_key_regenerated.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+[metadata]
+creation_date = "2020/08/19"
+ecs_version = ["1.5.0"]
```
Comment (threat-punter) [inline rules/azure/credential_access_storage_account_key_regenerated.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

File: rules/azure/credential_access_storage_account_key_regenerated.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+maturity = "production"
+updated_date = "2020/08/19"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a rotation to storage account access keys in Azure. Regenerating access keys can affect any applications
+or Azure services that are dependent on the storage account key.
```
Comment (threat-punter) [inline rules/azure/credential_access_storage_account_key_regenerated.toml:11]: I'd recommend including a sentence to explain why an adversary might perform this action.

**Review by threat-punter — APPROVED** [review]
LGTM. I made a minor suggestion to include a little more info in the description.

File: rules/azure/credential_access_storage_account_key_regenerated.toml (hunk 2 above)
Comment (bm11100) [reply rules/azure/credential_access_storage_account_key_regenerated.toml]: 👍

**Pull request merged**